src/python/dxpy/scripts/dx.py

```diff
@@ -2522,26 +2522,49 @@ def wait(args):
 def build(args):
     sys.argv = ['dx build'] + sys.argv[2:]
 
+    def get_from_exec_desc(args):
+        """
+        Return source executable description when --from option is used
+        
+        """
+        exec_describe_fields={'fields':{"properties":True, "details":True},'defaultFields':True}
+        _, _, exec_result = try_call(resolve_existing_path,
+                                     args._from,
+                                     expected='entity',
+                                     ask_to_resolve=False,
+                                     expected_classes=["applet", "workflow"],
+                                     all_mult=False,
+                                     allow_mult=False,
+                                     describe=exec_describe_fields)
+
+        if exec_result is None:
+            err_exit('Could not resolve {} to an existing applet or workflow.'.format(args._from), 3)
+        elif len(exec_result)>1:
+            err_exit('More than one match found for {}. Please use an applet/workflow ID instead.'.format(args._from), 3)
+        else:
+            if exec_result[0]["id"].startswith("applet"):
+                return exec_result[0]["describe"]
+            elif exec_result[0]["id"].startswith("workflow"):
+                return exec_result[0]["describe"]
```

Comment:
Since `expected_classes` argument allows only an applet or a workflow, we don't need the ifs here.

Reply:
Actually we still need this. The resolving function would not check the class of entities if the input is applet/workflow name instead of ID.  Then a corner case is we found something by name but it is not applet/workflow. But I can merge these two cases since they are returning the same thing.

Reply:
Got it. Thanks for the clarification.